spec/jobs/characterize_job_spec.rb

```diff
@@ -1,66 +1,31 @@
-      f.original_name = 'picture.png'
-      f.save!
-      allow(f).to receive(:save!)
-    end
-  end
+  include ActionDispatch::TestProcess
 
-  before do
-    allow(FileSet).to receive(:find).with(file_set_id).and_return(file_set)
-    allow(Hydra::Works::CharacterizationService).to receive(:run).with(file, filename)
-    allow(CreateDerivativesJob).to receive(:perform_later).with(file_set, file.id, filename)
-  end
-
-  context 'with valid filepath param' do
-    let(:filename) { File.join(fixture_path, 'world.png') }
-
-    it 'skips Hyrax::WorkingDirectory' do
-      expect(Hyrax::WorkingDirectory).not_to receive(:find_or_retrieve)
-      expect(Hydra::Works::CharacterizationService).to receive(:run).with(file, filename)
-      described_class.perform_now(file_set, file.id, filename)
+  describe ".perform" do
+    let(:char) { instance_double(Valkyrie::FileCharacterizationService) }
+    let(:user) { create(:user) }
+    let(:file_set) do
+      create_for_repository(:file_set,
+                            user: user,
+                            content: file)
     end
-  end
+    let(:file) { fixture_file_upload('/world.png', 'image/png') }
+    let(:file_id) { file_set.original_file.id }
 
-  context 'when the characterization proxy content is present' do
-    it 'runs Hydra::Works::CharacterizationService and creates a CreateDerivativesJob' do
-      expect(Hydra::Works::CharacterizationService).to receive(:run).with(file, filename)
-      expect(file).to receive(:save!)
-      expect(file_set).to receive(:update_index)
-      expect(CreateDerivativesJob).to receive(:perform_later).with(file_set, file.id, filename)
-      described_class.perform_now(file_set, file.id)
+    before do
+      allow(Valkyrie::FileCharacterizationService).to receive(:for).and_return(char)
+      allow(char).to receive(:characterize)
+      # allow(CreateDerivativesJob).to receive(:perform_later).with(file_set, file.id, filename)
     end
-  end
 
-  context 'when the characterization proxy content is absent' do
-    before { allow(file_set).to receive(:characterization_proxy?).and_return(false) }
-    it 'raises an error' do
-      expect { described_class.perform_now(file_set, file.id) }.to raise_error(StandardError, /original_file was not found/)
+    it "invokes Valkyrie::FileCharacterizationService" do
+      described_class.perform_now(file_set.id)
+      # because once in the factory
+      expect(char).to have_received(:characterize).twice
     end
-  end
 
-  context "when the file set's work is in a collection" do
-    let(:work)       { build(:work) }
-    let(:collection) { build(:collection) }
-
-    before do
-      allow(file_set).to receive(:parent).and_return(work)
-      allow(work).to receive(:in_collections).and_return([collection])
-    end
-    it "reindexes the collection" do
-      expect(collection).to receive(:update_index)
-      described_class.perform_now(file_set, file.id)
+    xit "invokes a CreateDerivativesJob" do
```

Comment:
Why are we skipping this test?  Should we just remove it?

Reply:
I have the call to CreateDerivativesJob commented out in the CharacterizeJob because it does not seem to be working at this point, and it causes the tests not to run at all since it gets called from the factory when the fileSet is created. I can certainly comment this out, just was thinking maybe it was easier to remember to uncomment it when the derivatives job is updated and working. It seems like something we should be testing.